File: Proyectos/TI.1 PAO202650_SINBDD_NRC30742/TI.1.4. PAO202650_RESTFUL_DOTNET_SINBDD_NRC30742/04. SERVIDOR/CONUNI_RESTFUL_DOTNET_GR5/CONUNI_RESTFUL_DOTNET_GR9S/Models/ConversionResponse.cs
namespace CONUNI_RESTFUL_DOTNET_GR9S.Models
{
    /// <summary>
    /// Modelo para las respuestas de conversión
    /// </summary>
    public class ConversionResponse
    {
        public double ValorOriginal { get; set; }
        public double ValorConvertido { get; set; }
        public string UnidadOrigen { get; set; }
        public string UnidadDestino { get; set; }
        public string Categoria { get; set; }
        public bool Exito { get; set; }
        public string Mensaje { get; set; }
        public long Timestamp { get; set; }
        
        public ConversionResponse()
        {
            Timestamp = System.DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
        }
        
        public ConversionResponse(double valorOriginal, double valorConvertido, 
                                string unidadOrigen, string unidadDestino, 
                                string categoria, bool exito, string mensaje)
        {
            ValorOriginal = valorOriginal;
            ValorConvertido = valorConvertido;
            UnidadOrigen = unidadOrigen;
            UnidadDestino = unidadDestino;
            Categoria = categoria;
            Exito = exito;
            Mensaje = mensaje;
            Timestamp = System.DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
        }
        
        /// <summary>
        /// Método estático para crear respuesta exitosa
        /// </summary>
        public static ConversionResponse CrearExito(double valorOriginal, double valorConvertido,
                                                   string unidadOrigen, string unidadDestino, string categoria)
        {
            return new ConversionResponse(valorOriginal, valorConvertido, unidadOrigen, 
                                        unidadDestino, categoria, true, "Conversión exitosa");
        }
        
        /// <summary>
        /// Método estático para crear respuesta de error
        /// </summary>
        public static ConversionResponse CrearError(string mensaje)
        {
            var response = new ConversionResponse();
            response.Exito = false;
            response.Mensaje = mensaje;
            return response;
        }
    }
}

File: Proyectos/TI.1 PAO202650_SINBDD_NRC30742/TI.1.3. PAO202650_RESTFUL_JAVA_SINBDD_NRC30742/04. SERVIDOR/CONUNI_RESTFUL_JAVA_GR05/src/main/java/ec/edu/monster/modelo/ConversionResponse.java
package ec.edu.monster.modelo;

import jakarta.xml.bind.annotation.XmlRootElement;

/**
 * Modelo para las respuestas de conversión
 * @author ACER NITRO V15
 */
@XmlRootElement
public class ConversionResponse {
    
    private double valorOriginal;
    private double valorConvertido;
    private String unidadOrigen;
    private String unidadDestino;
    private String categoria;
    private boolean exito;
    private String mensaje;
    private long timestamp;
    
    // Constructores
    public ConversionResponse() {
        this.timestamp = System.currentTimeMillis();
    }
    
    public ConversionResponse(double valorOriginal, double valorConvertido, 
                            String unidadOrigen, String unidadDestino, 
                            String categoria, boolean exito, String mensaje) {
        this.valorOriginal = valorOriginal;
        this.valorConvertido = valorConvertido;
        this.unidadOrigen = unidadOrigen;
        this.unidadDestino = unidadDestino;
        this.categoria = categoria;
        this.exito = exito;
        this.mensaje = mensaje;
        this.timestamp = System.currentTimeMillis();
    }
    
    // Método estático para crear respuesta exitosa
    public static ConversionResponse crearExito(double valorOriginal, double valorConvertido,
                                               String unidadOrigen, String unidadDestino, String categoria) {
        return new ConversionResponse(valorOriginal, valorConvertido, unidadOrigen, 
                                    unidadDestino, categoria, true, "Conversión exitosa");
    }
    
    // Método estático para crear respuesta de error
    public static ConversionResponse crearError(String mensaje) {
        ConversionResponse response = new ConversionResponse();
        response.setExito(false);
        response.setMensaje(mensaje);
        return response;
    }
    
    // Getters y Setters
    public double getValorOriginal() {
        return valorOriginal;
    }
    
    public void setValorOriginal(double valorOriginal) {
        this.valorOriginal = valorOriginal;
    }
    
    public double getValorConvertido() {
        return valorConvertido;
    }
    
    public void setValorConvertido(double valorConvertido) {
        this.valorConvertido = valorConvertido;
    }
    
    public String getUnidadOrigen() {
        return unidadOrigen;
    }
    
    public void setUnidadOrigen(String unidadOrigen) {
        this.unidadOrigen = unidadOrigen;
    }
    
    public String getUnidadDestino() {
        return unidadDestino;
    }
    
    public void setUnidadDestino(String unidadDestino) {
        this.unidadDestino = unidadDestino;
    }
    
    public String getCategoria() {
        return categoria;
    }
    
    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }
    
    public boolean isExito() {
        return exito;
    }
    
    public void setExito(boolean exito) {
        this.exito = exito;
    }
    
    public String getMensaje() {
        return mensaje;
    }
    
    public void setMensaje(String mensaje) {
        this.mensaje = mensaje;
    }
    
    public long getTimestamp() {
        return timestamp;
    }
    
    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }
    
    @Override
    public String toString() {
        return "ConversionResponse{" +
                "valorOriginal=" + valorOriginal +
                ", valorConvertido=" + valorConvertido +
                ", unidadOrigen='" + unidadOrigen + '\'' +
                ", unidadDestino='" + unidadDestino + '\'' +
                ", categoria='" + categoria + '\'' +
                ", exito=" + exito +
                ", mensaje='" + mensaje + '\'' +
                ", timestamp=" + timestamp +
                '}';
    }
}


